mesonbuild/backend/ninjabackend.py

```diff
@@ -1381,6 +1381,8 @@ def generate_dynamic_link_rules(self, outfile):
         ctypes = [(self.build.compilers, False)]
         if self.environment.is_cross_build():
             ctypes.append((self.build.cross_compilers, True))
+        else:
+            ctypes.append((self.build.cross_compilers, True))
```

Comment:
The bodies of both the true and false branches are identical. This seems suspicious....

Reply:
Seems like it.

Reply:
Unless you rather me fix the existing issue now, https://github.com/mesonbuild/meson/pull/4010 (which would be the second PR after this one) already rewrites all the code to deduplicate further and rule out this sort of thing for good.